doc/Programming-guide/Key-platform-features/Notifications/index.md

```diff
@@ -4,4 +4,803 @@ title: Notifications
+We also recommend that you review [collecting endpoint profiles guide]({{root_url}}Programming-guide/Key-platform-features/Data-collection) and [endpoint groups]({{root_url}}Programming-guide/Key-platform-features/Endpoint-groups) before you proceed with this guide.
+
+### Basic architecture
+
+The following diagram illustrates basic entities and data flows in scope of the notification management:
+
+* Notifications are generated based on the [notification schema](#notification-schema) created by the developer for the application 
+* The user or admin sends a notification via [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#TODO) call or using Administration UI (see [Sending notifications](#sending-notifications)). 
+
+![](images/basic_architecture_notification.png)
+
+### Configuring Kaa
+This section provides guidance on how to configure notifications in Kaa.
+
+#### Notification schema
+
+The Kaa notifications functionality allows transferring any data to endpoints. The structure of the data that is carried by notifications is defined by the notification schema configured on the Kaa server and built into Kaa endpoints. The notification schema is defined similarly to the endpoint profile schema. 
+In addition to the user-defined schema and its version, notifications are characterized by their type that can be either system or user. System notifications are processed by the prepackaged endpoint functions, while user notifications are handed over to the client code via the endpoint API.
+
+Since the data structure requirements may evolve throughout the Kaa-based system lifetime, the Kaa server can be configured to simultaneously handle notification schemas of multiple versions. 
+In this case, a notification will have multiple schema versions associated with it, as well as multiple sets of notification data so that each schema version is covered. To deliver such a notification to an endpoint, the server chooses the schema version supported by the endpoint.
+
+The default notification schema installed for Kaa applications is empty. For the purpose of this guide, we will use a simple notification schema shown in the following code block.
+
+```json
+{ 
+    "type":"record",
+    "name":"ExampleNotification",
+    "namespace":"org.kaaproject.kaa.schema.sample.notification",
+    "fields":[ 
+        { 
+            "name":"message",
+            "type":"string"
+        }
+    ]
+}
+```
+
+You can configure your own notification schema via [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#TODO) call or use Administration UI as shown below. 
+
+
+The list of notification schemas created by a tenant developer for the application is shown in the **Notification schemas** window.
+
+![Add Notification Schema 1](images/add_notification_schema_1.png)
```

Comment:
please rename TestNotifSchema in screenshoot to "Test Schema" or "Test Notification Schema"